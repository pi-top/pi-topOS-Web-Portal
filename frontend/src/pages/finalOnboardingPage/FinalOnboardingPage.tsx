import React from "react";

import Layout from "../../components/layout/Layout";

import rebootScreen from "../../assets/images/reboot-screen.png";
import styles from "./FinalOnboardingPage.module.css";

export const explanationMessage =
  "Press 'Finish' to complete the setup process";

export type Props = {
  onBackClick?: () => void;
  onNextClick?: () => void;
};

export default ({ onBackClick, onNextClick }: Props) => {
  return (
    <Layout
      banner={{
        src: rebootScreen,
        alt: "reboot-screen",
      }}
      prompt={
        <>
          Great, you're all set! <span className="green">Enjoy</span>!
        </>
      }
      explanation={explanationMessage}
      nextButton={{
        onClick: onNextClick,
        label: "Finish",
        disabled: false,
        hidden: false,
      }}
      backButton={{
        onClick: onBackClick,
        label: "Back",
        disabled: false,
        hidden: false,
      }}
      className={styles.root}
    ></Layout>
  );
};
