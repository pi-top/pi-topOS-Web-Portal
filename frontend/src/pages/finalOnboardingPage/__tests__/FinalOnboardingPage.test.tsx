import React from "react";
import {
  render,
  fireEvent,
  BoundFunction,
  GetByBoundAttribute,
  QueryByText,
  GetByText,
  QueryByBoundAttribute,
} from "@testing-library/react";

import FinalOnboardingPage, { Props } from "../FinalOnboardingPage";

describe("FinalOnboardingPage", () => {
  let defaultProps: Props;
  let finalOnboardingPage: HTMLElement;
  let getByAltText: BoundFunction<GetByBoundAttribute>;
  let queryByText: BoundFunction<QueryByText>;
  let getByText: BoundFunction<GetByText>;
  let queryByTestId: BoundFunction<QueryByBoundAttribute>;
  beforeEach(() => {
    defaultProps = {
      onBackClick: jest.fn(),
      onNextClick: jest.fn(),
    };

    ({
      container: finalOnboardingPage,
      getByAltText,
      queryByText,
      queryByTestId,
      getByText,
    } = render(<FinalOnboardingPage {...defaultProps} />));
  });

  it("renders correct image", () => {
    expect(getByAltText("final-screen")).toMatchSnapshot();
  });

  it("renders prompt correctly", () => {
    const prompt = finalOnboardingPage.querySelector(".prompt");
    expect(prompt).toMatchSnapshot();
  });

  it("doesn't render error message", () => {
    expect(queryByTestId("error-message")).not.toBeInTheDocument();
  });

  it("renders finish button", () => {
    expect(queryByText("Finish")).toBeInTheDocument();
  });

  it("renders back button", () => {
    expect(queryByText("Back")).toBeInTheDocument();
  });

  it("doesn't render skip button", () => {
    expect(queryByText("Skip")).not.toBeInTheDocument();
  });

  it("Back button isn't disabled", () => {
    expect(getByText("Back")).toHaveProperty("disabled", false);
  });

  it("Finish button isn't disabled", () => {
    expect(getByText("Finish").parentElement).toHaveProperty("disabled", false);
  });

  it("calls onBackClick on back button click", () => {
    fireEvent.click(getByText("Back"));

    expect(defaultProps.onBackClick).toHaveBeenCalled();
  });

  it("calls onNextClick on finish button click", () => {
    fireEvent.click(getByText("Finish"));

    expect(defaultProps.onNextClick).toHaveBeenCalled();
  });
});
