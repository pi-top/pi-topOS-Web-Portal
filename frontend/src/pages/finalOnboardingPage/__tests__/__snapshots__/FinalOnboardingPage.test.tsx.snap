// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinalOnboardingPage renders correct image 1`] = `
<img
  alt="reboot-screen"
  class="image"
  draggable="false"
  src="reboot-screen.png"
  style="max-height: 100%; max-width: 100%;"
/>
`;

exports[`FinalOnboardingPage renders prompt correctly 1`] = `
<h1
  class="prompt"
>
  Great, you're all set! 
  <span
    class="green"
  >
    Enjoy
  </span>
  !
</h1>
`;
